internal/consensus/reactor_test.go

```diff
@@ -789,7 +789,7 @@ func TestReactorValidatorSetChanges(t *testing.T) {
 	)
 	t.Cleanup(cleanup)
 
-	rts := setup(ctx, t, nPeers, states, 100) // buffer must be large enough to not deadlock
+	rts := setup(ctx, t, nPeers, states, nPeers+nVals) // buffer must be large enough to not deadlock
```

Comment:
doing this made it pass locally for me (and making the number larger made it pass less frequently), so I'm really just a dude with a wrench being empirical, and because it mostly fails in CI, you all get to play along too :D

Reply:
setting it to 1 makes it deadlock, setting it to 1000 makes it fail 10% or 20% of the time, the plus operation gives us 10, which seems to fail more like 1% of the time, which feels like an improvement. 

I suspect that the change to merge the sorting of channels slows or interleaves operations that were previously more independent, which exacerbates the flakiness of this test (which is structural, and due to the way that it relies on the event system).